app/cdap/components/GridTable/index.tsx

```diff
@@ -232,10 +182,16 @@ export default function GridTable() {
   return (
     <Box data-testid="grid-table">
       <BreadCrumb datasetName={wid} />
+      {Array.isArray(gridData?.headers) && gridData?.headers.length === 0 && (
+        <NoRecordScreen
+          title={T.translate('features.NoRecordScreen.gridTable.title')}
+          subtitle={T.translate('features.NoRecordScreen.gridTable.subtitle')}
+        />
+      )}
       <Table aria-label="simple table" className="test" data-testid="grid-table">
         <TableHead>
           <TableRow>
-            {headersNamesList?.length &&
+            {headersNamesList?.length > 0 &&
               headersNamesList.map((eachHeader) => (
```

Comment:
null and array check for headersNamesList

Reply:
not required

Reply:
Done